app/lib/Angel.coffee

```diff
@@ -296,51 +310,80 @@ module.exports = class Angel extends CocoClass
   simulateSync: (work) =>
     console?.profile? "World Generation #{(Math.random() * 1000).toFixed(0)}" if imitateIE9?
     work.t0 = now()
-    work.world = testWorld = new World work.userCodeMap
+    work.world = work.world = new World work.userCodeMap
+    work.world.synchronous = true
     work.world.levelSessionIDs = work.levelSessionIDs
     work.world.submissionCount = work.submissionCount
     work.world.fixedSeed = work.fixedSeed
     work.world.flagHistory = work.flagHistory ? []
-    work.world.difficulty = work.difficulty
-    work.world.loadFromLevel work.level
+    work.world.realTimeInputEvents = work.realTimeInputEvents ? []
+    work.world.difficulty = work.difficulty ? 0
+    work.world.loadFromLevel work.level, true
     work.world.preloading = work.preload
     work.world.headless = work.headless
     work.world.realTime = work.realTime
+    work.world.indefiniteLength = work.indefiniteLength;
+    work.world.justBegin = work.justBegin;
+    work.world.keyValueDb = work.keyValueDb;
     if @shared.goalManager
-      testGM = new GoalManager(testWorld)
-      testGM.setGoals work.goals
-      testGM.setCode work.userCodeMap
-      testGM.worldGenerationWillBegin()
-      testWorld.setGoalManager testGM
-    @doSimulateWorld work
+      goalManager = new GoalManager(work.world)
+      goalManager.setGoals work.goals
+      goalManager.setCode work.userCodeMap
+      goalManager.worldGenerationWillBegin()
+      work.world.setGoalManager goalManager
+    @beginSimulationSync work
+
+  beginSimulationSync: (work) ->
+    work.t1 = now()
+    #Math.random = work.world.rand.randf  # so user code is predictable - TODO: will this work when we do it on the main thread?
+    #Aether.replaceBuiltin('Math', Math)
+    #replacedLoDash = _.runInContext(window)
+    #_[key] = replacedLoDash[key] for key, val of replacedLoDash
```

Comment:
Are we not doing this work here? If we aren't, this commented out code should be removed.